docs/HACKING.md

```diff
@@ -143,3 +143,86 @@ $ make vmoverlay SKIP_INSTALL=1
 
 Of course, you can use this pattern for not just ostree but whatever else you'd
 like to install into the VM (e.g. bubblewrap, libsolv, etc...).
+
+## Using GDB with the rpm-ostree daemon
+
+If you're new to rpm-ostree, before using GDB, it may be helpful to review the
+[daemon architecture doc](../architecture-daemon.md) for an architecture recap.
+
+### Server-side (composes)
+
+Server-side composes do not use the daemon architecture and so one can naturally
+do e.g. `gdb --args rpm-ostree compose tree ...`. If using coreos-assembler, you
+can set the `COSA_RPMOSTREE_GDB` environment variable like this:
+
+```sh
+$ COSA_RPMOSTREE_GDB="gdb --args" cosa build
+```
+
+When cosa gets to the point of invoking rpm-ostree for the compose, it will call
+GDB instead.
+
+### Client-side
+
+On the client side, you need to use the `make vmsync` flow before using GDB
+because it also copies over the source files into `/root/sync`.
+
+For throwaway/fresh VMs, a simple approach is to just layer it using e.g.
+`rpm-ostree install gdb-minimal -A`, and then use it directly. (XXX:
+`apply-live` currently isn't compatible with `make vmsync`, so you'll want to
+reboot for now: https://github.com/ostreedev/ostree/issues/2369).
+
+It's also possible to use GDB from a privileged container. Make sure to use the
```

Comment:
There's a 3rd option which is https://developers.redhat.com/blog/2015/04/28/remote-debugging-with-gdb - I think we could even streamline this in coreos-assembler perhaps.

This also relates to https://github.com/coreos/coreos-assembler/issues/1645

Reply:
Definitely would be an interesting setup, though yeah we need to add sugar/tools there to make setting that up easier.